test/unit/shimmer.test.js

```diff
@@ -577,3 +582,242 @@ describe('shimmer', function () {
     })
   })
 })
+
+tap.test('Should not augment module when no instrumentation hooks provided', (t) => {
+  const agent = helper.instrumentMockedAgent()
+  t.teardown(() => {
+    helper.unloadAgent(agent)
+  })
+
+  const instrumentationOpts = {
+    moduleName: TEST_MODULE_PATH,
+    onError: () => {}
+  }
+  shimmer.registerInstrumentation(instrumentationOpts)
+
+  const loadedModule = require(TEST_MODULE_PATH)
+
+  t.equal(loadedModule.foo, 'bar')
+
+  const nrProps = Object.keys(loadedModule).filter((key) => {
+    return key.startsWith('__NR')
+  })
+
+  // Future proofing to catch any added symbols. If test  module modified to add own symbol
+  // will have to filter out here.
+  const nrSymbols = Object.getOwnPropertySymbols(loadedModule)
+
+  t.equal(nrProps.length, 0, `should not have any NR props but found: ${JSON.stringify(nrProps)}`)
+  t.equal(nrSymbols.length, 0, `should not have NR symbols but found: ${JSON.stringify(nrSymbols)}`)
+
+  t.end()
+})
+
+tap.test('Should not crash on empty instrumentation registration', (t) => {
+  const agent = helper.instrumentMockedAgent()
+  t.teardown(() => {
+    helper.unloadAgent(agent)
+  })
+
+  shimmer.registerInstrumentation()
```

Comment:
I'm fine doing this but I changed away from that because the summary output is worse. It just displays the function name. (Also tap, xunit and the like really just want you to throw). I do agree in general that explicit asserts can better communicate intent. Here there's only one assert so the test description makes obvious and so I opted for better output.

Current test summary output on failure:

```
 FAIL  ./test/unit/shimmer.test.js 1 failed of 55 462.472ms
 ✖ Cannot read properties of undefined (reading 'moduleName')
 ```
 
 Summary with `doesNotThrow()`:
 
 ```
  FAIL  ./test/unit/shimmer.test.js 1 failed of 55 477.539ms
 ✖ registerInstrumentation
 ```
 
 Anyways, fine changing just describing the 'why'.